src/app/shared/entity/external-machine-deployment.ts

```diff
@@ -13,33 +13,40 @@
 // limitations under the License.
 
 import {MachineDeployment} from '@shared/entity/machine-deployment';
-import {StatusIcon} from '@shared/utils/health-status';
+import {HealthStatus, StatusIcon} from '@shared/utils/health-status';
 import {AKSMachineDeploymentCloudSpec} from './provider/aks';
 
 export enum AKSMachineDeploymentMode {
   User = 'User',
   System = 'System',
 }
 
+export enum ExternalClusterMDState {
```

Comment:
Can we use `ExternalClusterState` enum since it contains all these values?

Reply:
its true, currently `ExternalClusterMDState` is a subset of `ExternalClusterState`. 
But I would recommend keeping them separate, to keep `Single Responsibility for data structures, as we don't want changes in one to affect the other.

Reply:
yes! i used the same enum but later on i updated due to following reasons.
- Handling MD state with `ExternalClusterState` enum which IMO, was not better 
- If we want to use same enum -> better to rename `ExternalClusterState` to `HealthStateStatus` to be more generic than? thoughts?
- Seperation of concernes.

If still want to be used same i can update it.

Reply:
We can keep it as it is for now